xds/src/main/java/io/grpc/xds/XdsCallCredentialsProvider.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2025 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import io.grpc.CallCredentials;
+import io.grpc.Internal;
+import java.util.Map;
+
+/**
+ * Provider of credentials data which will be propagated to the server for each RPC. The actual call
+ * credential to be used for a particular xDS communication will be chosen based on the bootstrap
+ * configuration.
+ */
+@Internal
+public abstract class XdsCallCredentialsProvider {
```

Comment:
I'm not wild about `io.grpc.xds.internal` (JwtTokenFileCallCredentials) depending on `io.grpc.xds` (XdsCallCredentialsProvider) which depends on `io.grpc.xds.internal` (XdsCallCredentialsRegistry). Can we move this class to internal, or the JWT stuff to io.grpc.xds?